mobile/lib/features/wallet/create_invoice_screen.dart

```diff
@@ -75,7 +75,7 @@ class _CreateInvoiceScreenState extends State<CreateInvoiceScreen> {
                         onPressed: amount == null
```

Comment:
🔧 The amount is optional! So skipping it should be ok. It is also reflected in the figma screens. The api should probably expect the an `Option` and in case of `None` you can leave the amount empty on the invoice.

Reply:
Three comments on this:
- I am not sure how to generate an amountless invoice from Rust
- The amount field input widget sets amount to 0 if you do not specify anh amount so that behaviour would need to change
- It might be a bit obscure (i.e hidden affordance) to be able to backspace the amount all the way to generate an amountless invoice. In figma this is a separate button but I left it off because of the first point that I mentioned here.

Reply:
> I am not sure how to generate an amountless invoice from Rust

I guess you can simply skip the `amount_milli_satoshi` in the invoice builder

https://github.com/get10101/10101/blob/808c6c28a2d7f348aa7ca1f106a2cc17ae018852/crates/ln-dlc-node/src/node.rs#L572-L596

> The amount field input widget sets amount to 0 if you do not specify anh amount so that behaviour would need to change

I guess, but you could also try to set the invoice amount to 0. I am not sure if this will be interpreted as `no amount set`